File: node.js/src/original/domain/books/entities/language.ts
import { DeepReadonly, NonEmptyString, NonEmptyUuid } from '#core/typing';

export type LanguageId = NonEmptyUuid<'LanguageId'>;
export type LanguageName = NonEmptyString<'LanguageName'>;

export type Language = DeepReadonly<{ Id: LanguageId; Name: LanguageName }>;

File: csharp/Slimmed/PublishingHouse.Domain/Books/Entities/Language.cs
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Books.Entities;

public record Language(LanguageId Id, LanguageName Name);
public record LanguageId(Guid Value): NonEmptyGuid(Value);
public record LanguageName(string Value): NonEmptyString(Value);

